samples/bot-tab-conversations/csharp/Source/ConversationalTabs.Web/ClientApp/src/routes/InquirySubEntityTab.tsx

```diff
@@ -0,0 +1,138 @@
+import { ReactNode, useEffect, useState } from 'react';
+import {
+  ArrowLeftIcon,
+  Button,
+  Flex,
+  Header,
+  Loader,
+} from '@fluentui/react-northstar';
+import { useNavigate, useParams } from 'react-router-dom';
+import { useQuery, useQueryClient } from 'react-query';
+import * as microsoftTeams from '@microsoft/teams-js';
+import { getSupportDepartment, getSupportDepartmentItem } from 'api';
+import { CustomerInquiryDetail } from 'components/CustomerInquiryDetail';
+import { ConsentRequest } from 'components/ConsentRequest';
+import { ApiErrorCode, CustomerInquiry, SupportDepartment } from 'models';
+import { isError } from 'utils/ErrorUtils';
+
+function InquirySubEntityTab() {
+  const [isChatOpen, setIsChatOpen] = useState<boolean>(false);
+  const [userHasConsented, setUserHasConsented] = useState<boolean>(false);
+  const [onLoadConversationOpened, setOnLoadConversationOpened] =
+    useState<boolean>(false);
+
+  const navigate = useNavigate();
+  const params = useParams();
+  const queryClient = useQueryClient();
+
+  const source: string = params.source ?? 'support-department';
+  const entityId: string = params.entityId ?? 'unknown';
+  const subEntityId: string = params.subEntityId ?? 'unknown';
+
+  useEffect(() => {
+    // When we navigate away, close the chat.
+    return () => closeConversation();
+  }, []);
+
+  const inquiry = useQuery<CustomerInquiry, Error>(
+    ['getSupportDepartmentItem', { entityId }, { subEntityId }],
+    () => getSupportDepartmentItem(entityId, subEntityId),
+    {
+      onSuccess: () => {
+        if (!onLoadConversationOpened) {
+          const successful = openConversation();
+          setOnLoadConversationOpened(successful);
+        }
+        setUserHasConsented(false);
+      },
+    },
+  );
+
+  const supportDepartment = useQuery<SupportDepartment, Error>(
+    ['getSupportDepartment', { entityId }],
+    () => getSupportDepartment(entityId),
+    {
+      onSuccess: () => {
+        if (!onLoadConversationOpened) {
+          const successful = openConversation();
+          setOnLoadConversationOpened(successful);
+        }
+        setUserHasConsented(false);
+      },
+    },
+  );
+
+  const openConversation = () => {
+    if (inquiry.data && supportDepartment.data) {
+      microsoftTeams.conversations.openConversation({
```

Comment:
# Open the conversation tab
With this conversation ID, we use the Teams SDK to open the conversation in the sidebar